File: Python/A_Project/B_Snake_and_Food/Utility/food.py
class Food:
    def __init__(self, food_positions):
        self.food_positions = food_positions
        self.current_food_index = 0

    def has_more(self) -> bool:
        return self.current_food_index < len(self.food_positions)

    def current(self):
        if not self.has_more():
            return None
        return self.food_positions[self.current_food_index]

    def consume(self):
        if self.has_more():
            self.current_food_index += 1

File: Ruby/A_Project/B_Snake_And_Food/Utility/food.rb
# frozen_string_literal: true

class Food
  def initialize(food_positions)
    @food_positions = food_positions
    @current_food_index = 0
  end

  def has_more
    @current_food_index < @food_positions.length
  end

  def current
    return nil unless has_more

    @food_positions[@current_food_index]
  end

  def consume
    @current_food_index += 1 if has_more
  end
end
